### ISSUE TITLE
Questions about Competition

### ISSUE THREAD

**Issue by eshau** [aws-neuron/nki-moe#15]
# Edit: Updated once @ 10:09 PM EST (Added Q5, 6)

# **Question 0:** Communication
Is there a better way to communicate with the AWS / Annapurna team? There are a lot of specific Trainium / NKI questions I know I and others will have over the next month and I think a faster line of communication vs Github issues would be nice. E.g., Slack or Discord.

# **Question 1:** Accuracy
From `CONTEST.md`:
> **Token Accuracy**: Logits must exactly match CPU reference
> Accuracy of generated output vs. our reference 

From `README.md`:
> implementation. Accuracy evaluation will be a binary assessor: Any benchmark that fails an accuracy threshold will result in a score of 0.

Is this strictly against the `bfloat16` version of `Qwen/Qwen3-30B-A3B`? Basically, I'm reading the Neuron / NKI documentation and I notice that Trainium 2/3 has an awful lot of FP8 tensor compute. Could we match against the `Qwen/Qwen3-30B-A3B-FP8` version and use `float8_e4m3` types instead? So I'm asking, do we:

1. Use only `bfloat16`, don't touch FP8 (unless we have some crazy algorithm/hack that allows us to match logits exactly)
2. Allow for comparison against `Qwen/Qwen3-30B-A3B-FP8` instead

# **Question 2:** `trn3.3xlarge` Availability
Will we get access to trn3s? There's a lot of difference between Trainium 2s and 3s and it's difficult to theoretically use the profiler.

# **Question 3:** Batch Size
From `CONTEST.md`:
> Performance Metrics (85%)
	**Throughput:** Tokens/second across batch sizes

From `README.md`:
> What batch size will be used in evalution? We will use a batch size of 1 in the final evaluation.

Just to be clear, which will it be?

# **Question 4:**  Evaluation Chip
From `README.md`:
> The contest organizers will execute each team's submission across the twenty withheld benchmarks on a dedicated Trainium instance.

I'm assuming this will be a singular Trainium3 chip / instance but just want to be sure.

# **Question 5:** Instruction Cost
From docs: https://awsdocs-neuron.readthedocs-hosted.com/en/latest/nki/guides/architecture/trainium_inferentia2_arch.html#arch-sec-vector-engine-perf
> There are a few exceptions to the above rule, depending on the data types and instruction type. See [NKI ISA API doc](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/nki/api/nki.isa.html) for instruction-specific instruction cost details.

I don't see it under there; is there somewhere we can find that?

# **Question 6:** GPSIMD Engine / CustomOps
My current understanding of GPSIMD is that we need to write custom C++ code with the ATen subset. However, the documentation looks a little old as it says in places it only supports NeuronCore-v2 (https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-customops/index.html#neuron-c-customops). Furthermore:

From `README.md`:
> Python Implementation
> - Uploaded Python file with your implementation of the model and kernels
> - NKI kernel implementations as Python modules containing custom NKI kernels for MoE operations
> - Integration code showing how NKI kernels replace baseline operators

Are we supposed to avoid using GPSIMD for this competition?
